orchestrator/packages/api/src/utils/grader-images.ts

```diff
@@ -1,20 +1,32 @@
-import { getConfig, GradingJobConfig } from "@codegrade-orca/common";
+import { getConfig, GradingJobConfig, logger } from "@codegrade-orca/common";
 import { execFile } from "child_process";
+import { existsSync } from "fs";
 import path = require("path");
 
 const CONFIG = getConfig();
 
+export class TouchGraderImageFileError extends Error {
+}
+
 export const touchGraderImageFile = ({
   grader_image_sha,
 }: GradingJobConfig): Promise<void> => {
   return new Promise((resolve, reject) => {
+    const tgzFileName = `${grader_image_sha}.tgz`;
+    const tgzPath = path.join(CONFIG.dockerImageFolder, tgzFileName);
+    if (!existsSync(tgzPath)) {
```

Comment:
this is not atomic; you have a potential TOCTOU violation here.  You should use `touch -c <theFile>` to say "update the modification time only if the file exists", and then subsequently check if the file exists (since there's a minuscule chance that the file could be deleted between this `existsSync` check and the subsequent `touch` check)